File: materia/networking/networking/ViewController.m
//
//  ViewController.m
//  networking
//
//  Created by Wallace Goncalves de Oliveira on 10/09/15.
//  Copyright © 2015 Wallace Goncalves de Oliveira. All rights reserved.
//

#import "ViewController.h"
#import "Connection.h"
#import "Session.h"
#import "AlamoFire.h"

@interface ViewController () {
    id<Networking> _net[3];
    NSArray<Networking> *_n;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _net[0] = [Connection new];
    _net[1] = [Session new];
    _net[2] = [AlamoFire new];
    
    // GET
    [_net[0] GET];
    [_net[1] GET];
    [_net[2] GET];
    
    // POST
    [_net[0] POST];
    [_net[1] POST];
    [_net[2] POST];
}

- (void)GET {
    
}

- (void)POST {
    
}

@end

File: materia/gps/gps/ViewController.swift
//
//  ViewController.swift
//  gps
//
//  Created by Wallace Goncalves de Oliveira on 23/09/15.
//  Copyright © 2015 Wallace Goncalves de Oliveira. All rights reserved.
//

import UIKit
import CoreLocation

class ViewController: UIViewController, CLLocationManagerDelegate {

    var manager = CLLocationManager();
    
//MARK: Actions
    
    @IBAction func ondeEstouTouched(sender: UIButton) {
        manager.requestLocation()
    }

//MARK: Lifecycle
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // ligamos o delegate
        manager.delegate = self
        
        // ativamos a autorizacao do usuario
        manager.requestAlwaysAuthorization()
        
        // determinamos a precisao do nosso GPS
        manager.desiredAccuracy = kCLLocationAccuracyHundredMeters
        
        // inicia o GPS -> Ativa uma leitura desenfreada das Lat e Long e
        // por isto consome muita bateria
        // manager.startUpdatingLocation()
    }
    
//MARK: CLLocationManagerDelegate
    
    func locationManager(manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
        var loca = locations.first
        
        if let myLoc = loca {
            print(myLoc.coordinate.latitude)
            print(myLoc.coordinate.longitude)
        }
    }
    
    func locationManager(manager: CLLocationManager, didFailWithError error: NSError) {
        print(error.description)
    }


}

